### PR TITLE
Add package for libnetcdf

### PR DESCRIPTION
**Pull request by rmcgibbo**

### CODE REVIEW

**Comment by Alexpux** [comment]
Can this packages be builded without dlfcn?

**Comment by rmcgibbo** [comment]
Yes, it can be, but a certain feature of the package then needs to be disabled.

**Comment by rmcgibbo** [comment]
I do not use that feature, so I dont if it is really important or not.

**Comment by rmcgibbo** [comment]
I updated the recipe to match the one in Arch Linux a little bit more closely. The full build log is here: https://ci.appveyor.com/project/rmcgibbo/mingw-packages/build/1.0.17

**Comment by rmcgibbo** [comment]
> Can this packages be builded without dlfcn?

@Alexpux would you prefer I change this to not use dlfcn?

**Comment by Alexpux** [comment]
Yeah

**Comment by rmcgibbo** [comment]
I will also try to get both static and shared libraries built, which requires #926

**Comment by rmcgibbo** [comment]
@Alexpux: Okay, updated not to use `dlfcn`. The build is happening now at https://ci.appveyor.com/project/rmcgibbo/mingw-packages/build/1.0.20

File: mingw-w64-netcdf/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  patch -p1 -i "${srcdir}"/0001-ocinternal-mkdir-patch.patch
+  patch -p1 -i "${srcdir}"/0002-onstack.patch
+}
+
+build() {
+  [[ -d ${srcdir}/build-static-${MINGW_CHOST} ]] && rm -rf ${srcdir}/build-static-${MINGW_CHOST}
+  mkdir -p ${srcdir}/build-static-${MINGW_CHOST} && cd ${srcdir}/build-static-${MINGW_CHOST}
+
+  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
+  cmake -G "MSYS Makefiles" \
```
Comment (Alexpux) [inline mingw-w64-netcdf/PKGBUILD]: Use full path to cmake command.
${MINGW_PREFIX}/bin/cmake

File: mingw-w64-netcdf/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	-DENABLE_DYNAMIC_LOADING=OFF \
+	-DENABLE_NETCDF_4=ON \
+	"${srcdir}/${_realname}-${pkgver}"
+  make || make
+
+  [[ -d ${srcdir}/build-shared-${MINGW_CHOST} ]] && rm -rf ${srcdir}/build-shared-${MINGW_CHOST}
+  mkdir -p ${srcdir}/build-shared-${MINGW_CHOST} && cd ${srcdir}/build-shared-${MINGW_CHOST}
+
+  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
+  cmake -G "MSYS Makefiles" \
```
Comment (Alexpux) [inline mingw-w64-netcdf/PKGBUILD]: As previous

File: mingw-w64-netcdf/PKGBUILD (hunk 1 above)
Comment (rmcgibbo) [inline mingw-w64-netcdf/PKGBUILD]: done.

File: mingw-w64-netcdf/0001-ocinternal-mkdir-patch.patch [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+diff --git a/oc2/ocinternal.c b/oc2/ocinternal.c
+index 0623e6a..3eb5205 100644
+--- a/oc2/ocinternal.c
++++ b/oc2/ocinternal.c
+@@ -610,7 +610,7 @@ ocset_curlproperties(OCstate* state)
+ 	char tmp[OCPATHMAX+1];
+         int stat;
+ 	snprintf(tmp,sizeof(tmp)-1,"%s/%s/",ocglobalstate.tempdir,OCDIR);
+-#ifdef _MSC_VER
++#if defined(_WIN32) || defined(_WIN64)
```
Comment (Alexpux) [inline mingw-w64-netcdf/0001-ocinternal-mkdir-patch.patch]: _WIN32 is defined for 64-bit. So you can leave here only _WIN32

Comment (rmcgibbo) [inline mingw-w64-netcdf/0001-ocinternal-mkdir-patch.patch]: Thanks. Done.

**Pull request merged**